### ISSUE TITLE
Set point temperature can show incorrectly if Eco Mode enabled

### ISSUE THREAD

**Issue by msp1974** [asantaga/wiserHomeAssistantPlatform#52]
Haven't noticed this before and don't know whether a software update from Wiser has made some changes, but when Eco mode enabled the CurrentSetPoint, which is used to display in the UI the target temp can be different to the schedule temp (not sure if true for manual set temps) and also to what is displayed in the app.

The app must use DisplayedSetPoint and therefore so should the custom_component.  Postman output below to show the issue.  See also setpointOrigin showing 'FromEcoIQ' - haven't seen this before either!

`{
    "id": 1,
    "ManualSetPoint": 230,
    "OverrideType": "EcoIQ",
    "OverrideSetpoint": 150,
    "ScheduleId": 1,
    "HeatingRate": 1200,
    "SmartValveIds": [
        4
    ],
    "UfhRelayIds": [],
    "Name": "Lounge",
    "Mode": "Auto",
    "DemandType": "Modulating",
    "WindowDetectionActive": false,
    "ControlSequenceOfOperation": "HeatingOnly",
    "HeatingType": "HydronicRadiator",
    "CalculatedTemperature": 233,
    "CurrentSetPoint": 150,
    "PercentageDemand": 0,
    "ControlOutputState": "Off",
    "WindowState": "Closed",
    "SetpointOrigin": "FromEcoIQ",
    "DisplayedSetPoint": 230,
    "ScheduledSetPoint": 230,
    "AwayModeSuppressed": false,
    "RoundedAlexaTemperature": 235,
    "EffectiveMode": "Auto",
    "PercentageDemandForItrv": 0
}`

**Comment by asantaga** [comment]
thanks this, im travelling for work at the moment. Does sound like this may have changed or currentSetPoint was the wrong one to use in the first place.

I'll dive deeper soon

**Comment by mrvladis** [comment]
I can definitely say that nothing has changed.

CurrentSetPoint - is the temperature that is currently set for the room as target. It can be equal to scheduled, to override or to away temperature.

ScheduledSetPoint - would always represent what temperature is currently set in schedule.

**Comment by Fulch** [comment]
Observed this different behaviour myself for the first time this evening when Eco mode switched on.

Pretty sure the firmware version number on all of my TRV's and the Roomstat has changed within the last few days to '42002', think it was previously something like '36001'.
Hub is on firmware '2.44.0' but I'm not sure if that has changed or not.

**Comment by scooper1** [comment]
According to wiser  a software update of hub has happened

Sent from Yahoo Mail on Android 
 
  On Thu, 23 Jan 2020 at 20:49, Fulch<notifications@github.com> wrote:   
Observed this different behaviour myself for the first time this evening when Eco mode switched on.

Pretty sure the firmware version number on all of my TRV's and the Roomstat has changed within the last few days to '42002', think it was previously something like '36001'.
Hub is on firmware '2.44.0' but I'm not sure if that has changed or not.

**Comment by asantaga** [comment]
This is now fixed in PR#54

Fix was  in climate.py
```
@property
    def target_temperature(self):
        target = (
            self.handler.get_hub_data()
            .getRoom(self.roomId)
            .get("CurrentSetPoint")
            / 10
        )
```
to 
```
    @property
    def target_temperature(self):
        target = (
            self.data.wiserhub
            .getRoom(self.room_id)
            .get("DisplayedSetPoint")
            / 10
        )
```
